clients/source.go

```diff
@@ -24,6 +24,8 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+var ErrMethodNotSupported = errors.New("method not supported")
```

Comment:
I don't think we should do that because but rather we need to use the grpc code directly. The CLI can have a helper function that check if a grpc is unimplemented and return some error message but the client shouldn't introduce a new error.

Reply:
Maybe I misunderstand the purpose of the client library, but my thinking is that it should abstract away the gRPC calls so that users of this library don't have to concern themselves with the underlying gRPC layer?

Reply:
Oh yeah, it's probably also worth mentioning that the gRPC errors don't support `errors.As`, otherwise it would have been okay to wrap the error before returning it: https://github.com/grpc/grpc-go/issues/2934. But now, since it's not supported, 

But as things stand, it's pretty tricky for callers of the client library to discover when it's not supported in a reliable way. Probably the best would be to look for a specific string in the error message.

Reply:
I see. ok I guess it's a nit. not sure right now what would be the best solution so let's skip it. mostly the things that worries me that it is now inconsistent because then we need to put it in all calls.